spec/validate.py

```diff
@@ -184,16 +240,12 @@ def validate_aql_on_arango(data):
             + f"  Extra params in schema: {params - query_bind_vars}")
 
 
-def _fatal(msg):
-    """Fatal error."""
-    sys.stderr.write(str(msg) + '\n')
-    sys.exit(1)
+if __name__ == '__main__':
 
+    validation_base_dir = None
+    if len(sys.argv) > 1:
+        validation_base_dir = sys.argv[1]
 
-if __name__ == '__main__':
     wait_for_arangodb()
-    try:
-        for s in ['data_source', 'stored_query', 'view', 'collection']:
-            validate_all(s)
-    except Exception as err:
-        _fatal(err)
+    n_errors = validate_all_by_type(validation_base_dir)
+    sys.exit(n_errors)
```

Comment:
Usually each exit code has a special meaning. And it is always limited to one byte, 0-255. So we want to have exit code 1 here.